### ISSUE TITLE
Version 2.0.0 crashing

### ISSUE THREAD

**Issue by veldechev** [yujitach/MenuMeters#120]
Not sure if this is the correct place to submit this, but the MenuMeter app (on Mac OS X 10.15.1 (19B88)) has been crashing lately with the following from the cosnsole: 

```
Application Specific Information:
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil'
terminating with uncaught exception of type NSException
abort() called
```

Not sure what triggered the exception.

**Comment by yujitach** [comment]
Could you post the full crash report, not just the application specific information?

**Comment by veldechev** [comment]
Here are the latest crash dumps from today. Thanks!

[crash_dumps.zip](https://github.com/yujitach/MenuMeters/files/3817784/crash_dumps.zip)

**Comment by yujitach** [comment]
Sorry for my silence; I was busy for a while. I haven't fixed the bug yet, but at least prepared a version with a better debug info. Could you use this and send me the crash report again?
[MenuMeters_2.0.1.zip](https://github.com/yujitach/MenuMeters/files/3853989/MenuMeters_2.0.1.zip)

**Comment by veldechev** [comment]
I updated to 2.0.1 (from the attached zip as auto update didn't work).

Now, am I supposed to just wait for the next time it crashes and collect the logs from the Console.app or will it write other trace files?

**Comment by yujitach** [comment]
Yes, please just send me the crash report as before. (2.0.1 was not made public, since no new feature is added.)

**Comment by veldechev** [comment]
Just a quick update - no crashes since I put 2.0.1 even though nothing has changed in this version... no other changes on other software whatsoever. what I did notice though is that I had an old (artifact) "greyed out" login item for MenuMeter in my account's preferences after I updated the version. I replaced the 2.0.0 app with 2.0.1 in Applications folder when installing. I'll report back with the crash dump if ever happens.

**Comment by yujitach** [comment]
Thanks for the updates. It's interesting but strange.

**Comment by veldechev** [comment]
Spoke too soon lol.
Here's the latest dump from shortly after I posted the comment above (what are the odds???)
[MenuMeters_2019-11-22-082530_vel-mbp.crash.zip](https://github.com/yujitach/MenuMeters/files/3880005/MenuMeters_2019-11-22-082530_vel-mbp.crash.zip)

**Comment by skyfaller** [comment]
I have MenuMeters v. 2.0.0 crashing whenever it opens at boot as a "Login Item", but it launches and continues to run fine when I open it manually. I now open it manually after I boot my computer, and manually delete the automatically created "Login Item", as a workaround.

**Comment by yujitach** [comment]
@skyfaller Could you attach the crash report to this issue?

**Comment by skyfaller** [comment]
I'm not sure I'm looking in the right place... when I search Console for menumeters after it fails to open on startup, all I see is diskarbitrationd saying "MenuMeters [411]:44379 not responding."

I guess I'll install 2.0.1 from the zip file in this thread, and try again.

**Comment by yujitach** [comment]
@skyfaller that's the same issue with #124 . I think I found the bug in the code. I'll update #124 later.

**Comment by yujitach** [comment]
I think I fixed it in the new version 2.0.2, available via auto-update. I tentatively close this issue. If it's not fixed, please reopen it.
